cloudmock/aws/mockec2/routetable.go

```diff
@@ -34,19 +51,138 @@ func (m *MockEC2) DescribeRouteTablesWithContext(aws.Context, *ec2.DescribeRoute
 }
 
 func (m *MockEC2) DescribeRouteTables(request *ec2.DescribeRouteTablesInput) (*ec2.DescribeRouteTablesOutput, error) {
-	if request.Filters != nil {
-		glog.Fatalf("filters not implemented: %v", request.Filters)
-	}
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+
+	glog.Infof("DescribeRouteTables: %v", request)
+
 	if request.DryRun != nil {
 		glog.Fatalf("DryRun not implemented")
 	}
-	if request.RouteTableIds != nil {
-		glog.Fatalf("RouteTableIds not implemented")
+
+	if len(request.RouteTableIds) != 0 {
+		request.Filters = append(request.Filters, &ec2.Filter{Name: s("route-table-id"), Values: request.RouteTableIds})
 	}
 
 	response := &ec2.DescribeRouteTablesOutput{}
+
 	for _, rt := range m.RouteTables {
-		response.RouteTables = append(response.RouteTables, rt)
+		allFiltersMatch := true
+		for _, filter := range request.Filters {
+			match := false
+			switch *filter.Name {
```

Comment:
Should we use the aws utility aws.String? Get the string safely.

Reply:
Probably yes, but we aren't really aiming for 100% fidelity and reliability.